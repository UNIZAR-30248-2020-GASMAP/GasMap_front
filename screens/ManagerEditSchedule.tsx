import * as React from 'react';
import { useState } from 'react';
import { Alert, Image, StyleSheet, ScrollView, TouchableOpacity, Switch } from 'react-native';

import { Text, View } from '../components/Themed';
import { Icon, Button } from 'react-native-elements';

import { servicesListToIcon } from '../models/listServicesToIcon'
import { Picker } from 'react-native';

import { postSchedule } from '../drivers/connection'



export default class ManagerEditSSchedule extends React.Component {

  constructor(props: any) {
    super(props)
    this.state = {
      //idGasolinera: this.props.route.params.datosGasolinera.id_gas,
      newDay: "Mon",
      newOpening: "00:00",
      newClosing: "00:00",
    };
  }

  render() {
    console.log("RENDER");
    return (
      <ScrollView>
        <View style={styles.container}>
          <View style={styles.containerTop}>
            <Text style={styles.mainTitle}>Your Gas Station is:</Text>
            <Text style={styles.plainBold}>
              <Icon
                name='location-on'
                type='material'
                color='black'
                size={30}
                onPress={() => console.log('hello')}
              />
            </Text>
            {/* Horizontal line */}
            <View
              style={{
                borderBottomColor: 'black',
                borderBottomWidth: 1,
              }}
            />
          </View>
          <View style={styles.containerDay}>
            <Text style={styles.plainBold}>
              Services Scheduler
            </Text>
          </View>
          <View style={styles.containerModal}>
            {/*Update day scheduler*/}
            <Text style={styles.plain}>
              Select day
            </Text>
            <Picker
            selectedValue={this.state.newDay}
            style={{ height: 20, width: 150, marginBottom: 22}}
            onValueChange={(itemValue, itemIndex) => this.setDay(itemValue)}
            >
            <Picker.Item label="Monday" value="Mon" />
            <Picker.Item label="Tuesday" value="Tue" />
            <Picker.Item label="Wednesday" value="Wed" />
            <Picker.Item label="Thursday" value="Thu" />
            <Picker.Item label="Friday" value="Fri" />
            <Picker.Item label="Saturday" value="Sat" />
            <Picker.Item label="Sunday" value="Sun" />
            </Picker>

            {/*Update opening scheduler*/}
            <Text style={styles.plain}>
              Select opening time
            </Text>
            <Picker
              selectedValue={this.state.newOpening}
              style={{ height: 20, width: 150, marginBottom: 22}}
              onValueChange={(itemValue, itemIndex) => this.setOpening(itemValue)}
              >
              <Picker.Item label="00:00" value="00:00" /><Picker.Item label="00:30" value="00:30" />
              <Picker.Item label="01:00" value="01:00" /><Picker.Item label="01:30" value="01:30" />
              <Picker.Item label="02:00" value="02:00" /><Picker.Item label="02:30" value="02:30" />
              <Picker.Item label="03:00" value="03:00" /><Picker.Item label="03:30" value="03:30" />
              <Picker.Item label="04:00" value="04:00" /><Picker.Item label="04:30" value="04:30" />
              <Picker.Item label="05:00" value="05:00" /><Picker.Item label="05:30" value="05:30" />
              <Picker.Item label="06:00" value="06:00" /><Picker.Item label="06:30" value="06:30" />
              <Picker.Item label="07:00" value="07:00" /><Picker.Item label="07:30" value="07:30" />
              <Picker.Item label="08:00" value="08:00" /><Picker.Item label="08:30" value="08:30" />
              <Picker.Item label="09:00" value="09:00" /><Picker.Item label="09:30" value="09:30" />
              <Picker.Item label="10:00" value="10:00" /><Picker.Item label="10:30" value="10:30" />
              <Picker.Item label="11:00" value="11:00" /><Picker.Item label="11:30" value="11:30" />
              <Picker.Item label="12:00" value="12:00" /><Picker.Item label="12:30" value="12:30" />
              <Picker.Item label="13:00" value="13:00" /><Picker.Item label="13:30" value="13:30" />
              <Picker.Item label="14:00" value="14:00" /><Picker.Item label="14:30" value="14:30" />
              <Picker.Item label="15:00" value="15:00" /><Picker.Item label="15:30" value="15:30" />
              <Picker.Item label="16:00" value="16:00" /><Picker.Item label="16:30" value="16:30" />
              <Picker.Item label="17:00" value="17:00" /><Picker.Item label="17:30" value="17:30" />
              <Picker.Item label="18:00" value="18:00" /><Picker.Item label="18:30" value="18:30" />
              <Picker.Item label="19:00" value="19:00" /><Picker.Item label="19:30" value="19:30" />
              <Picker.Item label="20:00" value="20:00" /><Picker.Item label="20:30" value="20:30" />
              <Picker.Item label="21:00" value="21:00" /><Picker.Item label="21:30" value="21:30" />
              <Picker.Item label="22:00" value="22:00" /><Picker.Item label="22:30" value="22:30" />
              <Picker.Item label="23:00" value="23:00" /><Picker.Item label="23:30" value="23:30" />
            </Picker>

            {/*Update closing scheduler*/}
            <Text style={styles.plain}>
              Select closing time
            </Text>
            <Picker
              selectedValue={this.state.newClosing}
              style={{ height: 20, width: 150, marginBottom: 22}}
              onValueChange={(itemValue, itemIndex) => this.setClosing(itemValue)}
              >
              <Picker.Item label="00:00" value="00:00" /><Picker.Item label="00:30" value="00:30" />
              <Picker.Item label="01:00" value="01:00" /><Picker.Item label="01:30" value="01:30" />
              <Picker.Item label="02:00" value="02:00" /><Picker.Item label="02:30" value="02:30" />
              <Picker.Item label="03:00" value="03:00" /><Picker.Item label="03:30" value="03:30" />
              <Picker.Item label="04:00" value="04:00" /><Picker.Item label="04:30" value="04:30" />
              <Picker.Item label="05:00" value="05:00" /><Picker.Item label="05:30" value="05:30" />
              <Picker.Item label="06:00" value="06:00" /><Picker.Item label="06:30" value="06:30" />
              <Picker.Item label="07:00" value="07:00" /><Picker.Item label="07:30" value="07:30" />
              <Picker.Item label="08:00" value="08:00" /><Picker.Item label="08:30" value="08:30" />
              <Picker.Item label="09:00" value="09:00" /><Picker.Item label="09:30" value="09:30" />
              <Picker.Item label="10:00" value="10:00" /><Picker.Item label="10:30" value="10:30" />
              <Picker.Item label="11:00" value="11:00" /><Picker.Item label="11:30" value="11:30" />
              <Picker.Item label="12:00" value="12:00" /><Picker.Item label="12:30" value="12:30" />
              <Picker.Item label="13:00" value="13:00" /><Picker.Item label="13:30" value="13:30" />
              <Picker.Item label="14:00" value="14:00" /><Picker.Item label="14:30" value="14:30" />
              <Picker.Item label="15:00" value="15:00" /><Picker.Item label="15:30" value="15:30" />
              <Picker.Item label="16:00" value="16:00" /><Picker.Item label="16:30" value="16:30" />
              <Picker.Item label="17:00" value="17:00" /><Picker.Item label="17:30" value="17:30" />
              <Picker.Item label="18:00" value="18:00" /><Picker.Item label="18:30" value="18:30" />
              <Picker.Item label="19:00" value="19:00" /><Picker.Item label="19:30" value="19:30" />
              <Picker.Item label="20:00" value="20:00" /><Picker.Item label="20:30" value="20:30" />
              <Picker.Item label="21:00" value="21:00" /><Picker.Item label="21:30" value="21:30" />
              <Picker.Item label="22:00" value="22:00" /><Picker.Item label="22:30" value="22:30" />
              <Picker.Item label="23:00" value="23:00" /><Picker.Item label="23:30" value="23:30" />
            </Picker>
            <View style={styles.containerButtons}>
            <Button
                containerStyle={styles.button}
                title="Save"
                onPress={() => {
                  // const replacedTime = "Mon: 08:00-23:00\\nTue: 07:00-23:00\\nWed: 08:00-23:00\\nThu: 09:00-23:00\\nFri: 10:00-23:00\\nSat: 06:00-00:00\\nSun: 06:00-00:00\\n"
                  const replacedTime = this.replaceTime(this.props.route.params.datosGasolinera.time_gas, 
                    this.state.newDay, this.state.newOpening, this.state.newClosing)
                  this.saveSchedule(replacedTime)
                }}
              />
              <Text>{"\t"}</Text>
              <Button
                containerStyle={styles.button}
                title="Cancel"
                onPress={() => {
                  Alert.alert("volver a la pantalla anterior")
                }}
              />
            </View>
          </View>
        </View>
      </ScrollView>
    );
  }
  
  setDay = (day: any) => {
    this.setState({
      newDay: day
    })
  }
  
  setOpening = (time: any) => {
    this.setState({
      newOpening: time
    })
  }

  setClosing = (time: any) => {
    this.setState({
      newClosing: time
    })
  }
  
  
  replaceTime = (oldSchedule: string, day: string, opening: string, closing: string) => {
    const exp = new RegExp(day + ": \\d\\d:\\d\\d\-\\d\\d:\\d\\d")
    const newSchedule = oldSchedule.replace(exp, day + ": " + opening + "-" + closing)
    return newSchedule
  }

  async saveSchedule(replacedTime: string) {
    await postSchedule(this.props.route.params.datosGasolinera.id_gas, 
      replacedTime)
      .then(data => {
      if (data == undefined) {
        Alert.alert('Connection error');
      } else {
        Alert.alert('New schedule updated');
        this.setState({ datosGasolinera: { ...this.props.route.params.datosGasolinera, time_gas: replacedTime } });
        this.props.navigation.navigate("ManagerGasStation", {datosGasolinera: { ...this.props.route.params.datosGasolinera, time_gas: replacedTime}});
      }
    })
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white'
  },
  containerTop: {
    alignItems: 'center',
    paddingBottom: '1%',
    backgroundColor: 'white'
  },
  containerDay: {
    alignItems: 'flex-start',
    backgroundColor: 'yellow'
  },
  mainTitle: {
    fontSize: 25,
    fontWeight: 'bold',
    paddingTop: '5%',
    paddingBottom: '1%',
    color: 'black',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    paddingTop: '0%',
    color: 'black',
  },
  textRow:{
    color: 'black',
    fontSize: 25,
    alignSelf:'center', 
    paddingTop: 10
  },
  plain: {
    fontSize: 17,
    fontWeight: 'bold',
    color: 'black',
  },
  plainBold: {
    fontSize: 23,
    color: 'black',
    fontWeight: "900",
    paddingBottom: '5%',
    paddingTop: '5%',
    paddingLeft: '5%'
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
  loginBtn: {
    width: "80%",
    backgroundColor: "#fb5b5a",
    borderRadius: 25,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 40,
    marginBottom: 10
  },
  loginText: {
    color: "white"
  },
  containerButtons:{
    backgroundColor: 'white',
    flexDirection: 'row',
    justifyContent: 'center'
  },
  button: {
    backgroundColor: "yellow",
    alignSelf: "center",
    width: "40%",
    borderRadius: 5,
    marginTop: 50,
    marginBottom: 50
  },
  containerModal:{
    marginTop: 22,
    alignItems: "center",
    backgroundColor: 'transparent',
  },
});
