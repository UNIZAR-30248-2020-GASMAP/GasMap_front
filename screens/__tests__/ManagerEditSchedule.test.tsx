import React from 'react';
import renderer from 'react-test-renderer';

import ManagerEditSchedule from '../ManagerEditSchedule';

test('Correct Rendered', () => {
    const tree = renderer.create(<ManagerEditSchedule />).toJSON();
    expect(tree).toMatchSnapshot()
});

test('Replace schedule', () => {
    const mngSchedule = new ManagerEditSchedule(null)
    const newOpening = '07:00'
    const newClosing = '23:30'
    const oldSchedule = "Mon: 00:00-00:00\\nTue: 00:00-00:00\\nWed: 00:00-00:00\\nThu: 00:00-00:00\\nFri: 00:00-00:00\\nSat: 00:00-00:00\\nSun: 00:00-00:00\\n"
    const newSchedule = "Mon: " + newOpening + "-" + newClosing + "\\nTue: 00:00-00:00\\nWed: 00:00-00:00\\nThu: 00:00-00:00\\nFri: 00:00-00:00\\nSat: 00:00-00:00\\nSun: 00:00-00:00\\n"
    
    const givenSchedule = mngSchedule.replaceTime(oldSchedule, 'Mon', '07:00', '23:30')
    expect(givenSchedule).toEqual(newSchedule)
});