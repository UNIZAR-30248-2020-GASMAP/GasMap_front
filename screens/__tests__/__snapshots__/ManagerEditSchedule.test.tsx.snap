// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct Rendered 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "paddingBottom": "1%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 25,
                "fontWeight": "bold",
                "paddingBottom": "1%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          Your Gas Station is:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 23,
                "fontWeight": "900",
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "yellow",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 23,
                "fontWeight": "900",
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          Services Scheduler
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "marginTop": 22,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 17,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Select day
        </Text>
        <View
          style={
            Object {
              "height": 20,
              "marginBottom": 22,
              "width": 150,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Monday",
                  "textColor": undefined,
                  "value": "Mon",
                },
                Object {
                  "label": "Tuesday",
                  "textColor": undefined,
                  "value": "Tue",
                },
                Object {
                  "label": "Wednesday",
                  "textColor": undefined,
                  "value": "Wed",
                },
                Object {
                  "label": "Thursday",
                  "textColor": undefined,
                  "value": "Thu",
                },
                Object {
                  "label": "Friday",
                  "textColor": undefined,
                  "value": "Fri",
                },
                Object {
                  "label": "Saturday",
                  "textColor": undefined,
                  "value": "Sat",
                },
                Object {
                  "label": "Sunday",
                  "textColor": undefined,
                  "value": "Sun",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 17,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Select opening time
        </Text>
        <View
          style={
            Object {
              "height": 20,
              "marginBottom": 22,
              "width": 150,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "00:00",
                  "textColor": undefined,
                  "value": "00:00",
                },
                Object {
                  "label": "00:30",
                  "textColor": undefined,
                  "value": "00:30",
                },
                Object {
                  "label": "01:00",
                  "textColor": undefined,
                  "value": "01:00",
                },
                Object {
                  "label": "01:30",
                  "textColor": undefined,
                  "value": "01:30",
                },
                Object {
                  "label": "02:00",
                  "textColor": undefined,
                  "value": "02:00",
                },
                Object {
                  "label": "02:30",
                  "textColor": undefined,
                  "value": "02:30",
                },
                Object {
                  "label": "03:00",
                  "textColor": undefined,
                  "value": "03:00",
                },
                Object {
                  "label": "03:30",
                  "textColor": undefined,
                  "value": "03:30",
                },
                Object {
                  "label": "04:00",
                  "textColor": undefined,
                  "value": "04:00",
                },
                Object {
                  "label": "04:30",
                  "textColor": undefined,
                  "value": "04:30",
                },
                Object {
                  "label": "05:00",
                  "textColor": undefined,
                  "value": "05:00",
                },
                Object {
                  "label": "05:30",
                  "textColor": undefined,
                  "value": "05:30",
                },
                Object {
                  "label": "06:00",
                  "textColor": undefined,
                  "value": "06:00",
                },
                Object {
                  "label": "06:30",
                  "textColor": undefined,
                  "value": "06:30",
                },
                Object {
                  "label": "07:00",
                  "textColor": undefined,
                  "value": "07:00",
                },
                Object {
                  "label": "07:30",
                  "textColor": undefined,
                  "value": "07:30",
                },
                Object {
                  "label": "08:00",
                  "textColor": undefined,
                  "value": "08:00",
                },
                Object {
                  "label": "08:30",
                  "textColor": undefined,
                  "value": "08:30",
                },
                Object {
                  "label": "09:00",
                  "textColor": undefined,
                  "value": "09:00",
                },
                Object {
                  "label": "09:30",
                  "textColor": undefined,
                  "value": "09:30",
                },
                Object {
                  "label": "10:00",
                  "textColor": undefined,
                  "value": "10:00",
                },
                Object {
                  "label": "10:30",
                  "textColor": undefined,
                  "value": "10:30",
                },
                Object {
                  "label": "11:00",
                  "textColor": undefined,
                  "value": "11:00",
                },
                Object {
                  "label": "11:30",
                  "textColor": undefined,
                  "value": "11:30",
                },
                Object {
                  "label": "12:00",
                  "textColor": undefined,
                  "value": "12:00",
                },
                Object {
                  "label": "12:30",
                  "textColor": undefined,
                  "value": "12:30",
                },
                Object {
                  "label": "13:00",
                  "textColor": undefined,
                  "value": "13:00",
                },
                Object {
                  "label": "13:30",
                  "textColor": undefined,
                  "value": "13:30",
                },
                Object {
                  "label": "14:00",
                  "textColor": undefined,
                  "value": "14:00",
                },
                Object {
                  "label": "14:30",
                  "textColor": undefined,
                  "value": "14:30",
                },
                Object {
                  "label": "15:00",
                  "textColor": undefined,
                  "value": "15:00",
                },
                Object {
                  "label": "15:30",
                  "textColor": undefined,
                  "value": "15:30",
                },
                Object {
                  "label": "16:00",
                  "textColor": undefined,
                  "value": "16:00",
                },
                Object {
                  "label": "16:30",
                  "textColor": undefined,
                  "value": "16:30",
                },
                Object {
                  "label": "17:00",
                  "textColor": undefined,
                  "value": "17:00",
                },
                Object {
                  "label": "17:30",
                  "textColor": undefined,
                  "value": "17:30",
                },
                Object {
                  "label": "18:00",
                  "textColor": undefined,
                  "value": "18:00",
                },
                Object {
                  "label": "18:30",
                  "textColor": undefined,
                  "value": "18:30",
                },
                Object {
                  "label": "19:00",
                  "textColor": undefined,
                  "value": "19:00",
                },
                Object {
                  "label": "19:30",
                  "textColor": undefined,
                  "value": "19:30",
                },
                Object {
                  "label": "20:00",
                  "textColor": undefined,
                  "value": "20:00",
                },
                Object {
                  "label": "20:30",
                  "textColor": undefined,
                  "value": "20:30",
                },
                Object {
                  "label": "21:00",
                  "textColor": undefined,
                  "value": "21:00",
                },
                Object {
                  "label": "21:30",
                  "textColor": undefined,
                  "value": "21:30",
                },
                Object {
                  "label": "22:00",
                  "textColor": undefined,
                  "value": "22:00",
                },
                Object {
                  "label": "22:30",
                  "textColor": undefined,
                  "value": "22:30",
                },
                Object {
                  "label": "23:00",
                  "textColor": undefined,
                  "value": "23:00",
                },
                Object {
                  "label": "23:30",
                  "textColor": undefined,
                  "value": "23:30",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 17,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Select closing time
        </Text>
        <View
          style={
            Object {
              "height": 20,
              "marginBottom": 22,
              "width": 150,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "00:00",
                  "textColor": undefined,
                  "value": "00:00",
                },
                Object {
                  "label": "00:30",
                  "textColor": undefined,
                  "value": "00:30",
                },
                Object {
                  "label": "01:00",
                  "textColor": undefined,
                  "value": "01:00",
                },
                Object {
                  "label": "01:30",
                  "textColor": undefined,
                  "value": "01:30",
                },
                Object {
                  "label": "02:00",
                  "textColor": undefined,
                  "value": "02:00",
                },
                Object {
                  "label": "02:30",
                  "textColor": undefined,
                  "value": "02:30",
                },
                Object {
                  "label": "03:00",
                  "textColor": undefined,
                  "value": "03:00",
                },
                Object {
                  "label": "03:30",
                  "textColor": undefined,
                  "value": "03:30",
                },
                Object {
                  "label": "04:00",
                  "textColor": undefined,
                  "value": "04:00",
                },
                Object {
                  "label": "04:30",
                  "textColor": undefined,
                  "value": "04:30",
                },
                Object {
                  "label": "05:00",
                  "textColor": undefined,
                  "value": "05:00",
                },
                Object {
                  "label": "05:30",
                  "textColor": undefined,
                  "value": "05:30",
                },
                Object {
                  "label": "06:00",
                  "textColor": undefined,
                  "value": "06:00",
                },
                Object {
                  "label": "06:30",
                  "textColor": undefined,
                  "value": "06:30",
                },
                Object {
                  "label": "07:00",
                  "textColor": undefined,
                  "value": "07:00",
                },
                Object {
                  "label": "07:30",
                  "textColor": undefined,
                  "value": "07:30",
                },
                Object {
                  "label": "08:00",
                  "textColor": undefined,
                  "value": "08:00",
                },
                Object {
                  "label": "08:30",
                  "textColor": undefined,
                  "value": "08:30",
                },
                Object {
                  "label": "09:00",
                  "textColor": undefined,
                  "value": "09:00",
                },
                Object {
                  "label": "09:30",
                  "textColor": undefined,
                  "value": "09:30",
                },
                Object {
                  "label": "10:00",
                  "textColor": undefined,
                  "value": "10:00",
                },
                Object {
                  "label": "10:30",
                  "textColor": undefined,
                  "value": "10:30",
                },
                Object {
                  "label": "11:00",
                  "textColor": undefined,
                  "value": "11:00",
                },
                Object {
                  "label": "11:30",
                  "textColor": undefined,
                  "value": "11:30",
                },
                Object {
                  "label": "12:00",
                  "textColor": undefined,
                  "value": "12:00",
                },
                Object {
                  "label": "12:30",
                  "textColor": undefined,
                  "value": "12:30",
                },
                Object {
                  "label": "13:00",
                  "textColor": undefined,
                  "value": "13:00",
                },
                Object {
                  "label": "13:30",
                  "textColor": undefined,
                  "value": "13:30",
                },
                Object {
                  "label": "14:00",
                  "textColor": undefined,
                  "value": "14:00",
                },
                Object {
                  "label": "14:30",
                  "textColor": undefined,
                  "value": "14:30",
                },
                Object {
                  "label": "15:00",
                  "textColor": undefined,
                  "value": "15:00",
                },
                Object {
                  "label": "15:30",
                  "textColor": undefined,
                  "value": "15:30",
                },
                Object {
                  "label": "16:00",
                  "textColor": undefined,
                  "value": "16:00",
                },
                Object {
                  "label": "16:30",
                  "textColor": undefined,
                  "value": "16:30",
                },
                Object {
                  "label": "17:00",
                  "textColor": undefined,
                  "value": "17:00",
                },
                Object {
                  "label": "17:30",
                  "textColor": undefined,
                  "value": "17:30",
                },
                Object {
                  "label": "18:00",
                  "textColor": undefined,
                  "value": "18:00",
                },
                Object {
                  "label": "18:30",
                  "textColor": undefined,
                  "value": "18:30",
                },
                Object {
                  "label": "19:00",
                  "textColor": undefined,
                  "value": "19:00",
                },
                Object {
                  "label": "19:30",
                  "textColor": undefined,
                  "value": "19:30",
                },
                Object {
                  "label": "20:00",
                  "textColor": undefined,
                  "value": "20:00",
                },
                Object {
                  "label": "20:30",
                  "textColor": undefined,
                  "value": "20:30",
                },
                Object {
                  "label": "21:00",
                  "textColor": undefined,
                  "value": "21:00",
                },
                Object {
                  "label": "21:30",
                  "textColor": undefined,
                  "value": "21:30",
                },
                Object {
                  "label": "22:00",
                  "textColor": undefined,
                  "value": "22:00",
                },
                Object {
                  "label": "22:30",
                  "textColor": undefined,
                  "value": "22:30",
                },
                Object {
                  "label": "23:00",
                  "textColor": undefined,
                  "value": "23:00",
                },
                Object {
                  "label": "23:30",
                  "textColor": undefined,
                  "value": "23:30",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "backgroundColor": "white",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "yellow",
                "borderRadius": 5,
                "marginBottom": 50,
                "marginTop": 50,
                "overflow": "hidden",
                "width": "40%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#2089dc",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Save
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                undefined,
              ]
            }
          >
            	
          </Text>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "yellow",
                "borderRadius": 5,
                "marginBottom": 50,
                "marginTop": 50,
                "overflow": "hidden",
                "width": "40%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#2089dc",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Cancel
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
